# Projeto-BioinformÃ¡tica

# ğŸ§¬ Protein Annotation Pipeline

This repository contains a **modular** and **reproducible** protein annotation pipeline implemented in **Snakemake**, integrating tools based on **homology**, **structure**, and **machine learning**.

---

## ğŸ“ Project Structure

```plaintext
project/
â”œâ”€â”€ config/
â”‚   â””â”€â”€ config.yaml                  # Pipeline configuration
â”œâ”€â”€ data/
â”‚   â”œâ”€â”€ databases/                   # Tool databases
â”‚   â””â”€â”€ proteins/
â”‚       â”œâ”€â”€ example.fasta            # Example FASTA file
â”‚       â””â”€â”€ trunc_fasta.py           # Script to truncate the proteins name
â”œâ”€â”€ results/                         # Outputs organized by tool/module
â”œâ”€â”€ workflow/
â”‚   â”œâ”€â”€ envs/
â”‚   â”‚   â”œâ”€â”€ module_1/                # Conda environments for module 1 (homology)
â”‚   â”‚   â”œâ”€â”€ module_2/                # Conda environments for module 2 (structure)
â”‚   â”‚   â””â”€â”€ module_3/                # Conda environments for module 3 (machine learning)
â”‚   â”œâ”€â”€ rules/
â”‚   â”‚   â”œâ”€â”€ module_1/                # Snakemake rules for module 1 (homology)
â”‚   â”‚   â”œâ”€â”€ module_2/                # Snakemake rules for module 2 (structure)
â”‚   â”‚   â””â”€â”€ module_3/                # Snakemake rules for module 3 (machine learning)
â””â”€â”€ Snakefile                        # Snakefile           
```

---

## âš™ï¸ Requisitos

- Linux / WSL
- Snakemake
- Conda

---

## ğŸš€ MÃ³dulos da Pipeline

### 1. Homology-based annotation
- Tools: `reCOGnizer`, `UPIMAPI`, `eggNOG-mapper`

### 2. Structure-based annotation
- Tools: `ColabFold`, `FoldSeek`, `DeepFRI`

### 3. Machine learning-based annotation
- Tools: `DeepGO2`, `Clean EC`

---

## ğŸ”§ Installation
```bash
git clone https://github.com/B-Neil/Projeto-Bioinform-tica.git
cd Projeto-Bioinform-tica/Project

```
## â–¶ï¸ Usage

1. Place your protein sequences in a FASTA file under `data/proteins/` (e.g., `example.fasta`)
2. Edit `config/config.yaml` to set the right parameters
3. Run the pipeline using: 

```bash
#N = Number of cores to be used
snakemake --use-conda --cores N --conda-frontend conda
